### PR TITLE
Use ghcr.io to store closure

### PR DESCRIPTION
**Pull request by hamishmack**

### CODE REVIEW

File: static.nix [hunk 1]
```diff
@@ -69,6 +69,7 @@ pkgs.mkShell ({
     "--with-hsc2hs=${if pkgs.stdenv.hostPlatform.isAarch64 then "aarch64" else "x86_64"}-unknown-linux-musl-hsc2hs"
     # ensure that the linker knows we want a static build product
     "--enable-executable-static"
+    " "
```
Comment (angerman) [inline static.nix:72]: I think we can drop this?
```suggestion
```

File: flake.nix [hunk 2]
```diff
@@ -74,8 +74,8 @@
        supportedSystems = [
             "x86_64-linux"
             "x86_64-darwin"
-            "aarch64-linux"
-            "aarch64-darwin"
+#            "aarch64-linux"
+#            "aarch64-darwin"
```
Comment (angerman) [inline flake.nix:78]: We probably want to enable these again?

File: extra/benchmarks.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env nix-shell
```
Comment (angerman) [inline extra/benchmarks.py:1]: I guess we can drop this file for now.

File: bootstrap.nix [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# This is an experiment around the idea of statically evaluating and cache a
```
Comment (angerman) [inline bootstrap.nix:1]: Do we need this file right now?

File: .github/workflows/hello.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+          #- aarch64-darwin  
+          #- aarch64-linux 
+        compiler-nix-name:
+          #- ghc8107
+          - ghc927
+          - ghc961
+        target-platform:
+          #- ""
+          - "-static"
+          - "-js"
```
Comment (angerman) [inline .github/workflows/hello.yml:27]:
```suggestion
          - "-js"
          - "mingw32"
```

File: .github/workflows/gh-pages.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
+# TODO: the file should not be longer needed if we succeed to rely only on
```
Comment (angerman) [inline .github/workflows/gh-pages.yml:1]: I think this one can go as well.

File: .github/workflows/benchmarks.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (angerman) [inline .github/workflows/benchmarks.yml:1]: I'm not sure we need this either.

File: gh-pages/index.html [hunk 8]
```diff
@@ -0,0 +1 @@
+bootstrap.sh
```
Comment (angerman) [inline gh-pages/index.html:1]: I guess this can go?

File: gh-pages/bootstrap.sh [hunk 9]
```diff
@@ -0,0 +1,69 @@
+#! /bin/bash
```
Comment (angerman) [inline gh-pages/bootstrap.sh:1]: This one's also not needed right now. But might be useful later.

File: flake.nix [hunk 10]
```diff
@@ -4,41 +4,11 @@
...
-                meta = with lib; {
-                    description = "A modern and easy-to-use crypto library - VRF fork";
-                    homepage = "http://doc.libsodium.org/";
-                    license = licenses.isc;
-                    maintainers = [ "tdammers" "nclarke" ];
-                    platforms = platforms.all;
-                };
-            }) {};
-        };
+        inherit (iohk-nix.overlays) crypto;
```
Comment (angerman) [inline flake.nix:11]: This should be a separate PR. It has nothing to do with the ghcr.io closures.

**Comment by angerman** [comment]
I also think we should extract the windows-cross logic into its own PR. This PR is getting way too big, and commingles different topics :-/

File: tool-map.nix [hunk 11]
```diff
@@ -1,5 +1,13 @@
...
\ No newline at end of file
+let fixed-versions = { "hlint" = { "ghc8107" = { version = "3.3"; evalSystem = "x86_64-linux"; }; }; }; in
+compiler-nix-name: tool:
+  if tool == "haskell-language-server"
+    then {pkgs, ...}: rec {
+      src = pkgs.haskell-nix.sources."hls-1.10";
+      cabalProject = __readFile (src + "/cabal.project");
+      sha256map."https://github.com/pepeiborra/ekg-json"."7a0af7a8fd38045fd15fb13445bdcc7085325460" = "sha256-fVwKxGgM0S4Kv/4egVAAiAjV7QB5PBqMVMCfsv7otIQ=";
+      evalSystem = "x86_64-linux";
+    }
```
Comment (angerman) [inline tool-map.nix:12]: We need a comment here, as to why we do this. What's the logic/reason behind using _this_ approach? We should document that.

And it should probably be it's own PR as well :-/

File: flake.nix (hunk 10 above)
Comment (hamishmack) [reply flake.nix:11]: https://github.com/input-output-hk/devx/pull/41

File: .github/workflows/hello.yml [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          #- x86_64-darwin
+          - x86_64-linux
+          #- aarch64-darwin
+          #- aarch64-linux
```
Comment (angerman) [inline .github/workflows/hello.yml:19]: Do we want to test this for other compilers now as well?

File: cross-windows.nix [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+        )
+        # Not sure why this `unset` helps.  It might avoids some kind of overflow issue.  We see `wine` fail to start when building `cardano-wallet-cli` test `unit`.
+        unset pkgsHostTargetAsString
+        WINEDLLOVERRIDES="winemac.drv=d" WINEDEBUG=warn-all,fixme-all,-menubuilder,-mscoree,-ole,-secur32,-winediag WINEPREFIX=$TMP ${pkgs.pkgsBuildBuild.winePackages.minimal}/bin/wine64 $REMOTE_ISERV/${iserv-proxy-interpreter.exeName} tmp $PORT &
+        (>&2 echo "---| ${iserv-proxy-interpreter.exeName} should have started on $PORT")
+        RISERV_PID="$!"
+        ${iserv-proxy}/bin/iserv-proxy $@ 127.0.0.1 "$PORT"
+        (>&2 echo "---> killing ${iserv-proxy-interpreter.exeName}...")
+        kill $RISERV_PID
+      '';
```
Comment (angerman) [inline cross-windows.nix:55]: This is logic from haskell.nix I believe? could we use it from there directly, without replicating it here?

File: tool-map.nix [hunk 14]
```diff
@@ -15,5 +15,6 @@ compiler-nix-name: tool:
       cabalProject = __readFile (src + "/cabal.project");
       # sha256 for a `source-repository-package` in the `cabal.project` file.
       sha256map."https://github.com/pepeiborra/ekg-json"."7a0af7a8fd38045fd15fb13445bdcc7085325460" = "sha256-fVwKxGgM0S4Kv/4egVAAiAjV7QB5PBqMVMCfsv7otIQ=";
+      evalSystem = "x86_64-linux";
     }
-  else fixed-versions.${tool}.${compiler-nix-name} or "latest"
\ No newline at end of file
+  else fixed-versions.${tool}.${compiler-nix-name} or { evalSystem = "x86_64-linux"; }
```
Comment (angerman) [inline tool-map.nix]: This needs a comment.

File: cross-windows.nix (hunk 13 above)
Comment (hamishmack) [reply cross-windows.nix]: I would like to replace the use of `pkgs.mkShell` in `devx` with something like `(pkgs.haskell-nix.tool compiler-nix-name "hello" {}).project.shellFor {}`.  Then we could also share things like the `cabal` wrappers.  Is there a reason we have not done that?

In this particular case we would need to make this same fix in haskell.nix since it does not wrap `ghc` (haskell.nix just passes setup `--ghc-options` flags to cabal when building new components).

**Comment by angerman** [comment]
@hamishmack so we can't do aarch64-darwin and aarch64-linux, because we need to _build_ on those architectures in GHA?

File: cross-windows.nix (hunk 13 above)
Comment (hamishmack) [reply cross-windows.nix]: I'm going to mark this as resolved for now as `cross-windows.nix` is going to go in a future PR anyway.

Comment (angerman) [reply cross-windows.nix]: The `mkShell` is likely due to trying to get the least complicated thing working. Trying to keep it really simple. If we learn that we need a more complex solution, we can likely go for that. We should however leave a comment justifying the decision.

**Review by angerman — APPROVED** [review]
LGTM, though I'm. a bit confused about the tools module
